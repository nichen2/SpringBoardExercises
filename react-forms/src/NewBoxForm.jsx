import { useState } from 'react'

function NewBoxForm({addBox}) {

  const INITIAL_STATE = {
    width: "",
    height: "",
    color: ""
  }
  const [formData, setFormData] = useState(INITIAL_STATE)
  
  const handleChange = e => {
    const { name, value} = e.target;
    setFormData(data => ({
        ...data,
        [name]: value
    }))
  }

  const handleSubmit = e => {
    e.preventDefault();
    addBox({
        width: formData.width,
        height: formData.height,
        backgroundColor: formData.color,
    });
    setFormData(INITIAL_STATE);
  }

  return (
    <>
      <form onSubmit={handleSubmit}>
        <label htmlFor='width'>Width</label>
        <input
            id="width"
            type="text"
            name="width"
            placeholder="width"
            value={formData.width}
            onChange={handleChange}
        />
        <label htmlFor='height'>Height</label>
        <input
            id="height"
            type="text"
            name="height"
            placeholder="height"
            value={formData.height}
            onChange={handleChange}
        />
        <label htmlFor='color'>Background Color</label>
        <input
            id="color"
            type="text"
            name="color"
            placeholder="color"
            value={formData.color}
            onChange={handleChange}
        />
        <button>Add New Box</button>
      </form>
    </>
  )
}

export default NewBoxForm
