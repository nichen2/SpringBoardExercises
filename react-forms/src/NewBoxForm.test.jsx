import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

describe('NewBoxForm Component', () => {
  it('renders without crashing', () => {
    render(<NewBoxForm />);
  });

  it('includes all form fields', () => {
    render(<NewBoxForm />);
    expect(screen.getByLabelText(/width/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/height/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/background color/i)).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});
