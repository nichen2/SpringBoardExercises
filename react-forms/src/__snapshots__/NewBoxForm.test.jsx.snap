// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewBoxForm Component > matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      name="width"
      placeholder="width"
      type="text"
      value=""
    />
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      name="height"
      placeholder="height"
      type="text"
      value=""
    />
    <label
      for="color"
    >
      Background Color
    </label>
    <input
      id="color"
      name="color"
      placeholder="color"
      type="text"
      value=""
    />
    <button>
      Add New Box
    </button>
  </form>
</DocumentFragment>
`;
